### ISSUE TITLE
Filtering who to list in widget

### ISSUE THREAD

**Issue by eighty20results** [sbrajesh/bp-members-with-uploaded-avatar-widget#3]
Hi,

I was looking for a filter to use so I can filter in specific groups in the widget. As far as I can tell, there's no immediate way to affect the user query you issue?

Would you mind adding a filter to the code so developers could add code to only list the users/groups/whatever they decided they needed to show?

**Comment by sbrajesh** [comment]
Hi,
Thank you for asking.
Will it work for you if I add a filter on query parameters here?
https://github.com/sbrajesh/bp-members-with-uploaded-avatar-widget/blob/master/bp-member-with-uploaded-avatar.php#L206

**Comment by eighty20results** [comment]
I was thinking that if you added a filter on the result itself, we could simply unset the results we didn't want (so the filter would be here):

https://github.com/sbrajesh/bp-members-with-uploaded-avatar-widget/blob/master/bp-member-with-uploaded-avatar.php#L214

**Comment by sbrajesh** [comment]
I have no issues in adding that but that will be wastage of one Query

**Comment by eighty20results** [comment]
I guess if the filter was for the query arguments themselves, it would probably. I'm looking to limit users shown based on their membership level in a different plugin, and worst case I could simply provide an array of user IDs for the include parameter.

**Comment by sbrajesh** [comment]
Well then, I will add two filters . Will update here in an hour after changes.

**Comment by sbrajesh** [comment]
HI,
I have added 2 filters. One for query args and one for filtering users. Please check and let me know if that will work for you.

Thank you
Brajesh

**Comment by eighty20results** [comment]
Thanks. I was looking at the code. Wondering if you may need a filter here too?

https://github.com/sbrajesh/bp-members-with-uploaded-avatar-widget/blob/master/bp-member-with-uploaded-avatar.php#L257

**Comment by sbrajesh** [comment]
Thanks. That is not required as we are applying the filter on the final result, it does not matter from where it is coming.

**Comment by eighty20results** [comment]
True. Just a thought, but wondering if it's not potentially confusing if someone uses the `bp_member_with_uploaded_avatar_query_args` filter and not the `bp_member_with_uploaded_avatar_users` filter and expects the result to be filtered by their rules/arguments/?

**Comment by sbrajesh** [comment]
No, I don't think so.
Both have specific purpose. The _users filter allows filtering on available user ids and may be useful in some cases like explicitly removing/adding some users.

The args parameter is for those who need full control over the query and want to modify the query based on some xprofile field or some meta.

**Comment by eighty20results** [comment]
Right. But, as far as I can tell, you'd only allow that level of control if they want the users who have avatars (with the current filters). If they want the users who do not have avatars as well, they wouldn't get that level of control? 

By the way, thanks for being so incredibly responsive!

**Comment by sbrajesh** [comment]
No problem.

You are right about that. The goal of this widget is to promote Members With Avatars. The options to include all was added later(and as you might have seen, needs different query) on the request of a few BuddyDev members.  I am assuming that most of the people using it are interested in the original use case and not for just listing all members.

Though I can add another filter, I will prefer to not do that unless you or someone really needs that.
Please do let me know if you need to filter that too.
